charts/redis-ha/templates/_configs.tpl

```diff
@@ -161,23 +221,44 @@
 
     find_master() {
         echo "Verifying redis master.."
-        echo "  ping (${MASTER}:${REDIS_PORT})"
+        if [ "$REDIS_PORT" -ne 0 ]; then
+            echo "  ping (${MASTER}:${REDIS_PORT})"
+        else
+            echo "  ping (${MASTER}:${REDIS_TLS_PORT})"
+        fi
         echo "  $(date).."
         if [ "$(redis_ping_retry 3)" != "PONG" ]; then
             echo "  $(date) Can't ping redis master (${MASTER})"
             echo "Attempting to force failover (sentinel failover).."
-            echo "  on sentinel (${SERVICE}:${SENTINEL_PORT}), sentinel grp (${MASTER_GROUP})"
-            echo "  $(date).."
-            if redis-cli -h "${SERVICE}" -p "${SENTINEL_PORT}" sentinel failover "${MASTER_GROUP}" | grep -q 'NOGOODSLAVE' ; then
-                echo "  $(date) Failover returned with 'NOGOODSLAVE'"
-                echo "Setting defaults for this pod.."
-                setup_defaults
-                return 0
-            fi
+            
+            if [ "$SENTINEL_PORT" -ne 0 ]; then
+                echo "  on sentinel (${SERVICE}:${SENTINEL_PORT}), sentinel grp (${MASTER_GROUP})"
+                echo "  $(date).."
+                if redis-cli -h "${SERVICE}" -p "${SENTINEL_PORT}" {{ if .Values.sentinel.auth }} -a "${SENTINELAUTH}"{{ end }} sentinel failover "${MASTER_GROUP}" | grep -q 'NOGOODSLAVE' ; then
+                    echo "  $(date) Failover returned with 'NOGOODSLAVE'"
+                    echo "Setting defaults for this pod.."
+                    setup_defaults
+                    return 0
+                fi
```

Comment:
See previous comment about TLS being assumed. I think we should check if TLS is enabled AND a port for TLS is defined, rather then just looking for Sentinel port.